source/platform/renderers/OpenGLCore/OpenGLCore.cpp

```diff
@@ -0,0 +1,743 @@
+
+            // Missing image, not loaded properly?
+            if (!tex)
+                return DrawMissingImage(rect);
+
+            Translate(rect);
+
+            if (m_currentTexture != *tex || m_activeProgram != m_fontProgram)
+            {
+                Flush();
+                glActiveTexture(GL_TEXTURE0);
+                glBindTexture(GL_TEXTURE_2D, *tex);
+                m_currentTexture = *tex;
+                m_activeProgram = m_fontProgram;
+            }
+
+            AddVert(rect.x,        rect.y,        u1, v1);
+            AddVert(rect.x+rect.w, rect.y,        u2, v1);
+            AddVert(rect.x,        rect.y+rect.h, u1, v2);
+            AddVert(rect.x+rect.w, rect.y,        u2, v1);
+            AddVert(rect.x+rect.w, rect.y+rect.h, u2, v2);
+            AddVert(rect.x,        rect.y+rect.h, u1, v2);
+        }
+
+        void OpenGLCore::DrawTexturedRect(Gwk::Texture* texture, Gwk::Rect rect,
+                                          float u1, float v1, float u2, float v2)
+        {
+            auto* tex = (GLuint*)texture->data;
+
+            // Missing image, not loaded properly?
+            if (!tex)
+                return DrawMissingImage(rect);
+
+            Translate(rect);
+
+            if (m_currentTexture != *tex || m_activeProgram != m_texturedProgram)
+            {
+                Flush();
+                glActiveTexture(GL_TEXTURE0);
+                glBindTexture(GL_TEXTURE_2D, *tex);
+                m_currentTexture = *tex;
+                m_activeProgram = m_texturedProgram;
+            }
+
+            AddVert(rect.x,        rect.y,        u1, v1);
+            AddVert(rect.x+rect.w, rect.y,        u2, v1);
+            AddVert(rect.x,        rect.y+rect.h, u1, v2);
+            AddVert(rect.x+rect.w, rect.y,        u2, v1);
+            AddVert(rect.x+rect.w, rect.y+rect.h, u2, v2);
+            AddVert(rect.x,        rect.y+rect.h, u1, v2);
+        }
+
+        Gwk::Color OpenGLCore::PixelColor(Gwk::Texture* texture, unsigned int x, unsigned int y,
+                                      const Gwk::Color& col_default)
+        {
+            auto* tex = (GLuint*)texture->data;
+
+            if (!tex)
+                return col_default;
+
+            unsigned int iPixelSize = sizeof(unsigned char)*4;
+            glBindTexture(GL_TEXTURE_2D, *tex);
+            auto* data = new unsigned char[iPixelSize * texture->width * texture->height];
+            glGetTexImage(GL_TEXTURE_2D, 0, GL_RGBA, GL_UNSIGNED_BYTE, data);
+            unsigned int iOffset = (y*texture->width+x)*4;
+
+            Color c(data[0+iOffset], data[1+iOffset], data[2+iOffset], data[3+iOffset]);
+
+            // Retrieving the entire texture for a single pixel read
+            // is kind of a waste - maybe cache this pointer in the texture
+            // data and then release later on? It's never called during runtime
+            // - only during initialization.
```

Comment:
I plan to remove pixel sampling at some point. I'd like to have an editor, which just saves to file, and editor has colour picker.

Reply:
You meant theme editor, that will save theme with additional data?